src/dmd/objc.d

```diff
@@ -319,12 +465,55 @@ extern(C++) private final class Supported : Objc
 
     override void setObjc(ClassDeclaration cd)
     {
+        static bool isExtern(Dsymbols* members)
+        {
+            bool result = true;
+
+            members.foreachDsymbol((member) {
+                if (auto attrib = member.isAttribDeclaration)
+                    result = isExtern(attrib.decl);
+
+                else if (auto func = member.isFuncDeclaration)
+                {
+                    if (func.fbody)
+                    {
+                        result = false;
+                        return 1; // return !=0 to stop iteration
+                    }
+                }
+
+                return 0;
+            });
+
+            return result;
+        }
```

Comment:
You can eliminate the `result` variable by just checking `foreachDsymbol` for a non-zero result.

Reply:
Ok, but I think the code will be more difficult to read then.

Reply:
Fixed.